### DISCUSSION TITLE
TileLayer Error using Mapping (on_image_error)

### DISCUSSION THREAD

**Discussion by Michel7GitHub** [flet-dev/flet#4355] [Q&A]
### Duplicate Check

- [X] I have searched the [opened issues](https://github.com/flet-dev/flet/issues) and there are no duplicates

### Describe the bug

I use Windows 11 (Flet 0.24.1) and IOS mobiles (Ipad & Iphone) with Flet App version 0.24.1.
When I drag or zoom-in /zoom-out quickly, I have frequent errors from "Event handler for image error events." 
Once, I get an error, refreshing becomes quite slow.
I get the error on all environments : Windows, Iphone and Ipad.
I tried to use few different tile providers (OpenStreetMap, OnTopoMap, Google, MapTile) and have same problem with all.
However, using OpenTopoMap gives more frequent errors.

```
import flet as ft
import flet.map as map

def main(page: ft.Page):

    def handle_event(e: map.MapEvent):
        if e.source == map.MapEventSource.DRAG_END:
            print("Drag_END ............. \n", e.source)
            
    page.add(
        map.Map(
            expand=True,
            configuration=map.MapConfiguration(
                initial_center=map.MapLatitudeLongitude(27.66, -15.58),
                initial_zoom=10,
                interaction_configuration=map.MapInteractionConfiguration(
                    flags=map.MapInteractiveFlag.ALL
                ),
                on_event=handle_event
            ),
            layers=[
                map.TileLayer(
                    #url_template="http://mt1.google.com/vt/lyrs=p&hl=en&x={x}&y={y}&z={z}",
                    url_template="https://tile.openstreetmap.org/{z}/{x}/{y}.png",
                    url_template="http://c.tile.opentopomap.org/{z}/{x}/{y}.png",
                    on_image_error=lambda e: print("TileLayer Error"),
                ),
                map.MarkerLayer(
                    markers=[
                        map.Marker(
                            content=ft.Icon(ft.icons.LOCATION_ON),
                            coordinates=map.MapLatitudeLongitude(15, 10), rotate=False
                        ),
                    ],
                )
            ],
        ),
    )

ft.app(target=main, view=ft.AppView.WEB_BROWSER)
```

### Code sample

<details open><summary>Code</summary>

```python
[Paste your code here]
```

</details>

### To reproduce

Try the code I sent.
Quickly Drag / zoom-in zoom-out in different regions of the map.
You should have an error...

### Screenshots / Videos

<details open>
<summary>Captures</summary>

[Upload media here]

</details>

### Operating System

Windows

### Operating system details

Windows 11

### Flet version

0.24.1

### Regression

I'm not sure / I don't know

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

**Comment by ndonkoHenri** [comment]
I guess zooming in and out quickly hinders proper/timely fetching and display of the map tiles.

Can you give the pre-release a try? We made some enhancements, which might hopefully fix this.
`pip install flet --pre -U `

**Comment by Michel7GitHub** [comment]
Hi ndonkoHenri,
Sorry for this long delay. I was out of the country for 2 weeks.
First I installed pre-release using `pip install flet --pre -U`. 
I made a first test on Windows environment. I've got the following error : `TypeError: MapConfiguration.__init__() got an unexpected keyword argument 'on_event'`.
I removed `'on_event'` line just to see if pre-release can solve the `TileLayer Error`  ... but It doesn't change the problem I have.
I made a second test on my iPhone. I can't see map (blank screen)!
Conclusion, pre-release caused a new problem.
If I can help, let me know.
Regards,
Michel
